src/malli/generator.cljc

```diff
@@ -127,15 +127,24 @@
   (let [{:keys [min max]} (-min-max schema options)
         child (-> schema m/children first)
         gen (generator child options)]
-    (gen/fmap f (if (-unreachable-gen? gen)
-                  (if (<= (or min 0) 0 (or max 0))
-                    (gen/return [])
-                    (-never-gen options))
-                  (gen/vector-distinct gen {:min-elements min, :max-elements max, :max-tries 100})))))
+    (if (-unreachable-gen? gen)
+      (if (= 0 (or min 0))
+        (gen/return (f []))
+        (-never-gen options))
+      (gen/fmap f (gen/vector-distinct gen {:min-elements min, :max-elements max, :max-tries 100
+                                            :ex-fn #(ex-info (str "Could not generate enough distinct elements for schema "
```

Comment:
Malli follow a "fail fast with data, pretty errors in dev-mode" philosophy. This enables both smaller bundle sizes on ClojureScript and fast runtime in prod (things like `pr-str` are really slow).

e.g. 

```clojure
(m/schema :enum)
```
fails with just enough data, while:

```clojure
((requiring-resolve 'malli.dev/start!))

(m/schema :enum)
```

pretty prints the proper text of what happened.


Would it be possible to do same here? Currently, the `malli.dev/start!` hi-jacks actual throwing of error, I guess you can't control that here, just generating the exception, to be throwed by `test.check`?

Reply:
Thanks for the feedback.

I think it's just a matter of calling `m/-exception`, I'll look into it.

Reply:
Could you take another look?